core/src/main/java/fr/sncf/osrd/envelope_sim/allowances/mareco_impl/CoastingGenerator.java

```diff
@@ -76,7 +73,7 @@ public static EnvelopePart coastFromEnd(
 
         assert constrainedBuilder.getLastPos() < endPos;
 
-        if (!reachedLowLimit && constrainedBuilder.getLastPos() != envelope.getBeginPos())
+        if (constrainedBuilder.getLastPos() != envelope.getBeginPos())
```

Comment:
@multun @axrolld I'll need you to double check this: why did we need to re-compute the coasting curve when we reach the low speed limit? This causes float inaccuracies that are hard to avoid and it doesn't seem necessary.

Reply:
it very much makes sense: you compute things backward first, and if you hit a lower bound, it means computing it forward is required, as it would not have the same result

Reply:
True, I misunderstood what happened in this case.

Well it's annoying. I don't see how it can be fixed without the eventual removal of floats here. Either we set the end speed and we have a position discontinuity, or we set the end position and we have a speed discontinuity.

I'll remove this commit, open a new issue for the bug not to forget it, and leave it there until the refacto. It sometimes causes 500 errors but it appears to be quite uncommon.

Reply:
I don't see how removing floats would fix the issue

Reply:
I tried this, but it moves the problem away, we still have asserts triggering in other parts of the code. In particular when building and handling the envelope. We'd need to add epsilons everywhere.